### PR TITLE
Replace command enums with command flag structs

### PR DESCRIPTION
**Pull request by InKryption**
Lays the groundwork for #47.

cc @ashpil

### CODE REVIEW

**Comment by ashpil** [comment]
Thanks! Pretty sure you could trivially add merge/intersect/etc support with the `FlagsMixin`.

**Comment by InKryption** [comment]
> Thanks! Pretty sure you could trivially add merge/intersect/etc support with the `FlagsMixin`.

I actually did originally attempt this, but it turned out to actually not be so trivial: due to having an uneven number of bits, the command flag structs aren't aligned to a power of two, and thus end up not being able to be `@bitCast`ed as the `FlagsMixin` methods would require (e.g. issues a compile error concerning the fact that `u87` has size 16, but `DeviceCommandFlags` has size 11, off the top of my head).

One solution to that that I ultimately decided wasn't worth the trouble was adding `_reserved_bit{d}` fields, but it made the code for re-ifying the `Dispatch` struct too unwieldy, and added way too much invariant checking in the `{}Wrapper` code. Since this is generated code, that might be OK, but it would be nice if the code there remained easy enough to refactor, which, this would make it not that.

Also can't align the first field e.g., since that's not allowed in packed structs anymore.

So it's looking like the best bet might be to just re-implement the methods - which I will try to do, but can't guarantee that I'll be very successful. So let's wait on that to merge this.

File: generator/vulkan/render.zig [hunk 1]
```diff
@@ -1029,21 +1010,22 @@ fn Renderer(comptime WriterType: type) type {
...
-                \\            for (std.enums.values({0s}Command)) |cmd_tag| {{
-                \\                if (@field(cmds, @tagName(cmd_tag))) {{
-                \\                    const PfnType = cmd_tag.PfnType();
+                \\            for (@typeInfo({0s}CommandFlags).Struct.fields) |field| {{
+                \\                if (@field(cmds, field.name)) {{
+                \\                    const title_case_name = [_]u8{{std.ascii.toUpper(field.name[0])}} ++ field.name[1..];
+                \\                    const PfnType = @field(vk, "Pfn" ++ title_case_name);
                 \\                    fields[i] = .{{
-                \\                        .name = cmd_tag.symbol(),
+                \\                        .name = "vk" ++ title_case_name,
```
Comment (Snektron) [inline generator/vulkan/render.zig:1028]: I guess this works but it does not feel like a very rigorous way to do it

**Review by Snektron — COMMENTED** [review]
Looks pretty good

> I actually did originally attempt this, but it turned out to actually not be so trivial:

One way to merge these two is to go the other way and use inline for for normal flags, though we'd still have problems with the toInt and fromInt function. I would say its not so bad to duplicate this code.

**Comment by Snektron** [comment]
Thanks a lot for the effort :)

**Pull request merged**